### PR TITLE
API add authorization and registration

### PR DESCRIPTION
**Pull request by BehindLoader**

### CODE REVIEW

File: api/src/models/account.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    required: true
+  }
+})
+
+schema.pre('save', function (next) {
+  this.password = crypto.createHash('md5')
+    .update(this.password)
+    .digest('hex')
+
+  next()
```
Comment (dannius) [inline api/src/models/account.js]: please use ; after each command. Looks ugly -_-

File: api/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  'extends': 'standard',
+  'globals': {
+    'Atomics': 'readonly',
+    'SharedArrayBuffer': 'readonly'
+  },
+  'parserOptions': {
+    'ecmaVersion': 2018,
+    'sourceType': 'module'
+  },
+  'rules': {
```
Comment (Gerasimov94) [inline api/.eslintrc.js]: empty object, fix this.

File: api/.gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+node_modules/
+database/
+
```
Comment (Gerasimov94) [inline api/.gitignore]: extra string, remove this, add .DS_Store for mac.

File: api/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Authorisation with existing user
+
+JSON data:
+* username {string}
+* password {string}
+
+Response:
+```
+{
+  "username": "YOUR_USERNAME",
```
Comment (Gerasimov94) [inline api/README.md]: i think it's unnecessary. Best way to add swagger, and create a JSON to describe all routes.

Comment (Gerasimov94) [reply api/README.md]: for this purpose swagger-ui-express is enough.

File: api/src/middlewares/error.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { ERRORS } from '../globals'
+
+export default function (err, req, res, next) {
+  let statusCode = 400
```
Comment (Gerasimov94) [inline api/src/middlewares/error.js]: add a ; rule for eslint

File: api/src/models/account.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import crypto from 'crypto'
+
```
Comment (Gerasimov94) [inline api/src/models/account.js]: libraries must be combined without empty strings

File: api/src/routes/account/auth.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken'
+
+import { ERRORS, JWT_SECRET } from '../../globals'
+import { Account } from '../../models'
+
+export default async function (req, res, next) {
+  const { username, password } = req.body
+
+  if (!username || !password) return next(ERRORS.MISSING_PARAM)
```
Comment (Gerasimov94) [inline api/src/routes/account/auth.js]: use a express-validator

**Pull request merged**